core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -403,6 +403,17 @@ private[spark] class BlockManager(
   }
 
   /**
+   * Cleanup code run in response to a failed local read.
+   * Must be called while holding a read lock on the block.
+   */
+  private def handleLocalReadFailure(blockId: BlockId): Nothing = {
+    releaseLock(blockId)
+    // Remove the missing block so that its unavailability is reported to the driver
+    removeBlock(blockId)
```

Comment:
No, I don't think so: internally, `removeBlock` acquires a write lock on the block, so if we called it before the `releaseLock` call then we'd be calling it while holding a read lock which would cause us to deadlock.